### PR TITLE
Expose stdout/stderr in Session

### PR DESCRIPTION
**Pull request by leonardoalt**
This PR:
- Makes explicit the notion that 0=stdin, 1=stdout, 2=stdout in the QueryCallback's "FS"
- Exposes the outputs in Session
- Removes printing to stdout and stderr in the callback itself. This is now the responsibility of the host if needed.
- Adds the Fibonacci test with stdout to CI using the write mechanism

The idea is that after this we should also expose the proof's publics and make a stream mechanism for inputs and outputs

### CODE REVIEW

File: pipeline/src/lib.rs [hunk 1]
```diff
@@ -59,7 +65,7 @@ impl HostContext {
                         as char;
                     match fd {
                         // stdin, stdout and stderr are supported by the default callback
-                        0..=2 => return Err(format!("Unsupported file descriptor: {fd}")),
+                        0 => return Err(format!("Unsupported file descriptor: {fd}")),
```
Comment (pacheco) [inline pipeline/src/lib.rs:68]: Comment right above needs to be updated i think.
Also looking at the code, who handles fds 1 and 2? is `HostContext` only used for riscv? 
`handle_simple_queries_callback` also seems to handle 1 and 2?

File: pipeline/src/lib.rs [hunk 2]
```diff
@@ -34,6 +34,12 @@ impl HostContext {
         (ctx, cb)
     }
 
+    pub fn clear_outputs(&mut self) {
```
Comment (pacheco) [inline pipeline/src/lib.rs:37]: shouldn't this be private and the pipeline then responsible for calling this when appropriate?

Comment (leonardoalt) [reply pipeline/src/lib.rs:37]: but will pipeline be able to call this if it's private?

**Comment by pacheco** [comment]
Also, what do we do about our `runtime::print`? it won't print anymore right?

File: pipeline/src/lib.rs (hunk 1 above)
Comment (leonardoalt) [reply pipeline/src/lib.rs:68]: IMO there should be no special treatment for 1 and 2 in the callback itself, the host should just read it afterwards from pipeline or session if needed

**Comment by leonardoalt** [comment]
> Also, what do we do about our `runtime::print`? it won't print anymore right?

we can print it in the test

**Comment by pacheco** [comment]
> > Also, what do we do about our `runtime::print`? it won't print anymore right?
> 
> we can print it in the test

i mean, won't it be confusing to users that `runtime::print` does nothing?

File: pipeline/src/lib.rs (hunk 2 above)
Comment (pacheco) [reply pipeline/src/lib.rs:37]: what I was thinking is if this should be "automatic" for pipeline users (as in, isn't it a kind of artifact too, that needs to be cleared if you run some part of the pipeline again)

**Comment by leonardoalt** [comment]
ahhh runtime print, sorry I confused with the test. Yea true... let me think about it

File: pipeline/src/lib.rs (hunk 1 above)
Comment (pacheco) [reply pipeline/src/lib.rs:68]: but there is code in `handle_simple_queries_callback` to handle it... is the `HostContext` callback always present? which one handles it?

File: pipeline/src/lib.rs (hunk 2 above)
Comment (leonardoalt) [reply pipeline/src/lib.rs:37]: it's kind of part of the witness, but in a different data structure. But I'm not sure I understand what the suggestion is

File: pipeline/src/lib.rs (hunk 1 above)
Comment (leonardoalt) [reply pipeline/src/lib.rs:68]: I removed that though, check `pipeline/src/lib.rs` here

File: pipeline/src/lib.rs [hunk 3]
```diff
@@ -169,22 +175,6 @@ pub fn handle_simple_queries_callback<'a, T: FieldElement>() -> impl QueryCallba
         let (id, data) = parse_query(query)?;
         match id {
             "None" => Ok(None),
-            "Output" => {
```
Comment (leonardoalt) [inline pipeline/src/lib.rs:172]: @pacheco here

Comment (pacheco) [reply pipeline/src/lib.rs:172]: oh right, my bad, i must have looked into the old version

**Review by pacheco — APPROVED** [review]

**Comment by leonardoalt** [comment]
I removed it from the queue again because somehow the newly added Fibonacci RISCV test (as an example) is running in every job, instead of in one.

**Review by pacheco — APPROVED** [review]

**Pull request merged**